backend/src/indexer/block.rs

```diff
@@ -265,6 +278,21 @@ impl PreparedBlock {
         }
         self.statistics.save(tx).await?;
         self.special_transaction_outcomes.save(tx).await?;
+
+        // process validator updates for staked amount and pool total staked amount
+        for (validator_id, validator_staking_information) in &self.validator_stake_updates {
```

Comment:
If you build up the information correctly, you can update these in a single sql query, which would perform better.
For inspiration look for places were we use `UNNEST`